courses/api.py

```diff
@@ -218,7 +219,8 @@ def create_run_enrollments(
             )
             if not created and not enrollment.active:
                 enrollment.edx_enrolled = edx_request_success
-                enrollment.edx_emails_subscription = True
+                subscribed = subscribe_to_edx_course_emails(user, run)
+                enrollment.edx_emails_subscription = True if subscribed else False
```

Comment:
do we want to update similar logic here as well? https://github.com/mitodl/mitxonline/blob/cc12b7556e44e6600eb0b7ba30043f077d2c1d5f/courses/views/v1/__init__.py#L220